From 6b7397699b5a356ecfb78497a26094b0fbbd334e Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Wed, 13 May 2015 10:43:41 +0100
Subject: [PATCH] MythDownloadManager: add some logging to try to track down
 the 100% cpu bug

This also allows the theme checker to be run once a minute rather than the
usual once an hour to try to trigger the bug quicker. Setting the
MYTHTV_DEBUGMDM environment variable will run the checker every minute.

So for 0.27.4-fixes use :-
MYTHTV_DEBUGMDM="1" mythfrontend -v file --loglevel=debug

Refs #12356

(cherry picked from commit 7e0441c38166ded77b3a1b3a869e40a5af389fa0)
---
 mythtv/libs/libmythbase/mythdownloadmanager.cpp |    8 ++++++++
 mythtv/programs/mythfrontend/themechooser.cpp   |   12 +++++++++++-
 2 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/mythdownloadmanager.cpp b/mythtv/libs/libmythbase/mythdownloadmanager.cpp
index 63e2e51..9a1983e 100644
--- a/mythtv/libs/libmythbase/mythdownloadmanager.cpp
+++ b/mythtv/libs/libmythbase/mythdownloadmanager.cpp
@@ -276,6 +276,8 @@ void MythDownloadManager::run(void)
             updateCookieJar();
         }
         m_infoLock->lock();
+        LOG(VB_FILE, LOG_DEBUG, LOC + QString("items downloading %1").arg(m_downloadInfos.count()));
+        LOG(VB_FILE, LOG_DEBUG, LOC + QString("items queued %1").arg(m_downloadQueue.count()));
         downloading = !m_downloadInfos.isEmpty();
         itemsInCancellationQueue = !m_cancellationQueue.isEmpty();
         m_infoLock->unlock();
@@ -297,9 +299,15 @@ void MythDownloadManager::run(void)
             m_queueWaitLock.lock();
 
             if (downloading)
+            {
+                LOG(VB_FILE, LOG_DEBUG, LOC + QString("waiting 200ms"));
                 m_queueWaitCond.wait(&m_queueWaitLock, 200);
+            }
             else
+            {
+                LOG(VB_FILE, LOG_DEBUG, LOC + QString("waiting for more items to download"));
                 m_queueWaitCond.wait(&m_queueWaitLock);
+            }
 
             m_queueWaitLock.unlock();
         }
diff --git a/mythtv/programs/mythfrontend/themechooser.cpp b/mythtv/programs/mythfrontend/themechooser.cpp
index a14beb0..b5950bc 100644
--- a/mythtv/programs/mythfrontend/themechooser.cpp
+++ b/mythtv/programs/mythfrontend/themechooser.cpp
@@ -1027,7 +1027,17 @@ ThemeUpdateChecker::ThemeUpdateChecker(void) :
     gCoreContext->SaveSetting("ThemeUpdateStatus", "");
 
     connect(m_updateTimer, SIGNAL(timeout()), SLOT(checkForUpdate()));
-    m_updateTimer->start(60 * 60 * 1000); // Run once an hour
+
+    if (getenv("MYTHTV_DEBUGMDM"))
+    {
+        LOG(VB_GENERAL, LOG_INFO, "Checking for theme updates every minute");
+        m_updateTimer->start(60 * 1000); // Run once a minute
+    }
+    else
+    {
+        LOG(VB_GENERAL, LOG_INFO, "Checking for theme updates every hour");
+        m_updateTimer->start(60 * 60 * 1000); // Run once an hour
+    }
 
     // Run once 15 seconds from now
     QTimer::singleShot(15 * 1000, this, SLOT(checkForUpdate()));
-- 
1.7.10.2

